lib/Macros/Sources/SwiftMacros/SwiftifyImportMacro.swift

```diff
@@ -557,6 +616,54 @@ public struct SwiftifyImportMacro: PeerMacro {
     return pointerParamIndex
   }
 
+  static func parseTypeMappingParam(_ paramAST: LabeledExprSyntax?) throws -> [String: String]? {
+    guard let unwrappedParamAST = paramAST else {
+      return nil
+    }
+    let paramExpr = unwrappedParamAST.expression
+    guard let dictExpr = paramExpr.as(DictionaryExprSyntax.self) else {
+      return nil
+    }
+    var dict : [String: String] = [:]
+    switch dictExpr.content {
+      case .colon(_):
+        return dict
+      case .elements(let types):
+        for element in types {
+          guard let key = element.key.as(StringLiteralExprSyntax.self) else {
+            throw DiagnosticError("expected a string literal, got '\(element.key)'", node: element.key)
+          }
+          guard let value = element.value.as(StringLiteralExprSyntax.self) else {
+            throw DiagnosticError("expected a string literal, got '\(element.value)'", node: element.value)
+          }
+          dict[key.representedLiteralValue!] = value.representedLiteralValue!
+        }
+      default:
+        throw DiagnosticError("unknown dictionary literal", node: dictExpr)
+    }
+    return dict
+  }
+
+  static func parseCxxSpanParams(
+    _ signature: FunctionSignatureSyntax,
+    _ typeMappings: [String: String]?
+  ) throws -> [ParamInfo] {
+    guard let typeMappings else {
+      return []
+    }
+    var result : [ParamInfo] = []
+    for (idx, param) in signature.parameterClause.parameters.enumerated() {
+      let typeName = try getTypeName(param.type).text;
+      if let desugaredType = typeMappings[typeName] {
+        if desugaredType.starts(with: "span") {
```

Comment:
I think this will cause an issue if any type alias from C/C++ where the desugared type starts with "span", right? It's probably a good idea to add a `cxxSpan` option to the `_SwiftifyInfo` enum.

Reply:
In the meantime, I am working on a follow-up PR. The content of the string we are matching against depends on how we generate the `typeMapping` in ClangImporter. Would you be OK if this part would stay as is and I only changed it in the follow up where this starts to work end to end?

Reply:
The follow-up PR where this logic changes: https://github.com/swiftlang/swift/pull/78422